sdks/python/src/opik/integrations/haystack/opik_span_bridge.py

```diff
@@ -0,0 +1,286 @@
+
+        This method is required by the Haystack Span interface.
+        Content tags are used for input/output data that should only be
+        logged when content tracing is enabled.
+
+        Args:
+            key: The content tag key (e.g., "component.input", "component.output").
+            value: The content tag value.
+        """
+        if not tracing.tracer.is_content_tracing_enabled:
+            return
+
+        if key.endswith(".input"):
+            self._convert_and_store_input_data(value)
+        elif key.endswith(".output"):
+            self._convert_and_store_output_data(value)
+
+        self._data[key] = value
+
+    def _convert_and_store_input_data(self, value: Any) -> None:
+        """
+        Convert Haystack input data to Opik format and store in span.
+
+        Args:
+            value: The input tag value to process.
+
+        Raises:
+            OpikException: If message conversion fails.
+        """
+        try:
+            if "messages" in value:
+                messages = [
+                    converters.convert_message_to_openai_format(message)
+                    for message in value["messages"]
+                ]
+                self._span_or_trace_data.update(input={"input": messages})
+            else:
+                self._span_or_trace_data.update(input={"input": value})
+        except Exception as e:
+            LOGGER.error("Failed to process input tag: %s", e, exc_info=True)
+            LOGGER.error(
+                "Failed to process input tag (type: %s, value: %r): %s",
+                type(value).__name__,
+                value,
+                e,
+                exc_info=True,
+            )
+            raise exceptions.OpikException(
+                f"Failed to process input tag (type: {type(value).__name__}, value: {value!r}): {e}"
+            ) from e
+
+    def _convert_and_store_output_data(self, value: Any) -> None:
+        """
+        Convert Haystack output data to Opik format and store in span.
+
+        Args:
+            value: The output tag value to process.
+
+        Raises:
+            OpikException: If message conversion fails.
+        """
+        try:
+            if "replies" in value:
+                if all(
+                    isinstance(reply, haystack_dataclasses.ChatMessage)
+                    for reply in value["replies"]
+                ):
+                    replies = [
+                        converters.convert_message_to_openai_format(message)
+                        for message in value["replies"]
+                    ]
+                else:
+                    replies = value["replies"]
+                self._span_or_trace_data.update(output={"replies": replies})
+            else:
+                self._span_or_trace_data.update(output={"output": value})
+        except Exception as e:
+            LOGGER.error("Failed to process output tag: %s", e, exc_info=True)
+            LOGGER.error(
```

Comment:
Do we need the two `LOGGER.error()` here with the same stack trace? It would be too much output which is almost identical.